core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -215,6 +215,19 @@ class TaskMetrics extends Serializable {
     inputMetrics.foreach(_.updateBytesRead())
   }
 
+  override def equals(other: Any): Boolean = other match {
+    case that: TaskMetrics => {
+      if (this.executorDeserializeTime != that.executorDeserializeTime) return false
+      if (this.executorRunTime != that.executorRunTime) return false
+      if (this.resultSize != that.resultSize) return false
+      if (this.jvmGCTime != that.jvmGCTime) return false
+      if (this.resultSerializationTime != that.resultSerializationTime) return false
+      if (this.hostname == null && that.hostname != null) return false
+      if (this.hostname != null) this.hostname.equals(that.hostname) else true
+    }
```

Comment:
it would be better if this looked like:
```
this.executorDeserializeTime == that.executorDeserializeTime &&
this.executorRunTime == that.executorRunTime &&
...
this.hostname == that.hostname
```
no need to check `equals` here because we're in scala